src/govuk/components/header/header.mjs

```diff
@@ -30,12 +48,23 @@ Header.prototype.init = function () {
  *
  * Sync the menu button class and the accessible state of the menu and the menu
  * button with the visible state of the menu
- *
- * @param {boolean} isVisible Whether the menu is currently visible
  */
-Header.prototype.syncState = function (isVisible) {
-  this.$menuButton.classList.toggle('govuk-header__menu-button--open', isVisible)
-  this.$menuButton.setAttribute('aria-expanded', isVisible)
+Header.prototype.syncState = function () {
+  if (this.mql === null || !this.mql.matches) {
+    this.$menuButton.removeAttribute('hidden')
+
+    this.$menuButton.classList.toggle('govuk-header__menu-button--open', this.menuIsOpen)
```

Comment:
As a thought (maybe outside of the scope of this work), would it be an idea to hook the styles onto a selector like `.govuk-header__menu-button[aria-expanded="true"]` and get rid of having to add/remove an additional class?

Reply:
This isn't something we do currently anywhere else in the codebase, but that's not a reason not to do it as it's a neat trick.